pkg/api-server/resource_endpoints.go

```diff
@@ -646,6 +658,92 @@ func (r *resourceEndpoints) matchingDataplanesForPolicy() restful.RouteFunction
 	}
 }
 
+func (r *resourceEndpoints) configForProxy() restful.RouteFunction {
+	return func(request *restful.Request, response *restful.Response) {
+		ctx := request.Request.Context()
+
+		name := request.PathParameter("name")
+		mesh, err := r.meshFromRequest(request)
+		if err != nil {
+			rest_errors.HandleError(ctx, response, err, "Failed to retrieve Mesh")
+			return
+		}
+		qparams, err := r.configForProxyParams(request)
+		if err != nil {
+			rest_errors.HandleError(ctx, response, err, "Failed to parse query parameters")
+			return
+		}
+
+		mc, err := r.meshContextBuilder.Build(ctx, mesh)
+		if err != nil {
+			rest_errors.HandleError(ctx, response, err, "Failed to build mesh context")
+			return
+		}
+
+		inspector, err := inspect.NewProxyConfigInspector(mc, r.zoneName, r.xdsHooks...)
+		if err != nil {
+			rest_errors.HandleError(ctx, response, err, "Failed to create proxy config inspector")
+			return
+		}
+
+		config, err := inspector.Get(ctx, name, *qparams.Shadow)
+		if err != nil {
+			rest_errors.HandleError(ctx, response, err, "Failed to inspect proxy config")
+			return
+		}
+
+		out := &api_types.InspectDataplanesConfig{
+			Xds: config,
+		}
+
+		if slices.Contains(*qparams.Include, api_types.Diff) {
+			currentConfig, err := inspector.Get(ctx, name, false)
+			if err != nil {
+				rest_errors.HandleError(ctx, response, err, "Failed to inspect current proxy config")
+				return
+			}
+			diff, err := inspect.Diff(config, currentConfig)
+			if err != nil {
+				rest_errors.HandleError(ctx, response, err, "Failed to compute diff")
+				return
+			}
+			out.Diff = &diff
+		}
+
+		if err := response.WriteAsJson(out); err != nil {
+			rest_errors.HandleError(ctx, response, err, "Failed writing response")
+		}
+	}
+}
+
+func (r *resourceEndpoints) configForProxyParams(request *restful.Request) (*api_types.GetMeshesMeshDataplanesNameConfigParams, error) {
+	params := &api_types.GetMeshesMeshDataplanesNameConfigParams{
+		Shadow:  pointer.To(false),
+		Include: &[]api_types.GetMeshesMeshDataplanesNameConfigParamsInclude{},
+	}
+
+	if shadow := request.QueryParameter("shadow"); shadow != "" {
```

Comment:
Do we need to support it in `kumactl` or it works out of the box?

Reply:
We need support in `kumactl`, I'm going to add this in the next PR